Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -729,19 +730,24 @@ void AddDatas(CFArrayRef fontDatas, CFArrayRef* errors) {
             if (data) {
                 if (CFSetContainsValue(m_fontDatasSet.get(), data)) {
                     __AppendErrorIfExists(outErrors, kCTFontManagerErrorAlreadyRegistered);
+                    ret = S_FALSE;
```

Comment:
why does this have to be S_FALSE?
Are you expecting this to succeed, where you have failures?